src/components/grid-view/GridView.js

```diff
@@ -54,33 +50,24 @@ class GridView extends React.Component<Props> {
     }
 
     cellRenderer = ({ dataKey, parent, rowIndex }: TableCellRendererParams) => {
-        const { columnCount, currentCollection, slotRenderer, onItemSelect } = this.props;
+        const { columnCount, currentCollection, slotRenderer } = this.props;
         const count = getProp(currentCollection, 'items.length', 0);
         const contents = [];
 
         const startingIndex = rowIndex * columnCount;
+        const maxSlotIndex = Math.min(startingIndex + columnCount, count);
 
-        for (let slotIndex = startingIndex; slotIndex < startingIndex + columnCount; slotIndex += 1) {
+        for (let slotIndex = startingIndex; slotIndex < maxSlotIndex; slotIndex += 1) {
             // using item's id as key is important for renrendering.  React Virtualized Table rerenders
             // on every 1px scroll, so using improper key would lead to image flickering in each
             // card of the grid view when scrolling.
-            let key;
-            let item = null;
-            if (currentCollection.items && currentCollection.items[slotIndex]) {
-                key = currentCollection.items[slotIndex].id;
-                item = currentCollection.items[slotIndex];
-            } else {
-                key = uniqueId('bdl-GridViewSlot');
-            }
-
             contents.push(
-                <GridViewSlot
-                    key={key}
-                    slotIndex={slotIndex}
-                    slotRenderer={slotIndex < count ? slotRenderer : null}
-                    item={item}
-                    onItemSelect={onItemSelect}
-                />,
+                <div
+                    key={getProp(currentCollection, `items[${slotIndex}].id`, uniqueId('bdl-GridView-slot'))}
```

Comment:
Minor, but maybe it would be better to wrap the default value in a anonymous function so that `uniqueId` isn't called each time even if it's not being used